File: src/main/java/kr/co/bookand/backend/feedback/controller/FeedbackController.java
package kr.co.bookand.backend.feedback.controller;

import io.swagger.annotations.Api;
import io.swagger.annotations.ApiOperation;
import kr.co.bookand.backend.common.domain.dto.PageResponse;
import kr.co.bookand.backend.feedback.domain.dto.FeedbackDto.*;
import kr.co.bookand.backend.feedback.service.FeedbackService;
import lombok.RequiredArgsConstructor;
import org.springframework.data.domain.Pageable;
import org.springframework.data.web.PageableDefault;
import org.springframework.http.ResponseEntity;
import org.springframework.web.bind.annotation.*;

@RestController
@RequestMapping("/api/v1/feedbacks")
@RequiredArgsConstructor
@Api(tags = "피드백 API")
public class FeedbackController {

    private final FeedbackService feedbackService;

    @ApiOperation(value = "피드백 생성")
    @PostMapping("")
    public ResponseEntity<FeedbackResponse> createFeedback(@RequestBody FeedbackRequest feedbackRequest) {
        return ResponseEntity.ok(feedbackService.createFeedback(feedbackRequest));
    }

    @ApiOperation(value = "피드백 리스트 조회")
    @GetMapping("")
    public ResponseEntity<PageResponse<FeedbackListResponse>> getFeedbackList(@PageableDefault Pageable pageable) {
        return ResponseEntity.ok(feedbackService.getFeedbackList(pageable));
    }

    @ApiOperation(value = "피드백 상세 정보 조회")
    @GetMapping("{feedbackId}")
    public ResponseEntity<FeedbackDetailResponse> getFeedbackDetail(@PathVariable Long feedbackId) {
        return ResponseEntity.ok(feedbackService.getFeedbackDetail(feedbackId));
    }
}

File: src/main/kotlin/kr/co/bookand/backend/feedback/controller/FeedbackController.kt
package kr.co.bookand.backend.feedback.controller

import io.swagger.annotations.Api
import io.swagger.annotations.ApiOperation
import kr.co.bookand.backend.account.service.AccountService
import kr.co.bookand.backend.feedback.dto.CreateFeedbackRequest
import kr.co.bookand.backend.feedback.dto.FeedbackIdResponse
import kr.co.bookand.backend.feedback.dto.FeedbackListResponse
import kr.co.bookand.backend.feedback.dto.FeedbackResponse
import kr.co.bookand.backend.feedback.service.FeedbackService
import kr.co.bookand.backend.issue.dto.IssueIdResponse
import org.springframework.data.domain.Pageable
import org.springframework.data.web.PageableDefault
import org.springframework.http.ResponseEntity
import org.springframework.web.bind.annotation.*


@RestController
@RequestMapping("/api/v1/feedbacks")
@Api(tags = ["피드백 API"])
class FeedbackController(
    private val feedbackService: FeedbackService,
    private val accountService : AccountService
){
    @ApiOperation(value = "피드백 생성")
    @PostMapping
    fun createFeedback(
        @RequestBody feedbackRequest: CreateFeedbackRequest
    ): ResponseEntity<FeedbackIdResponse> {
        val account = accountService.getCurrentAccount()
        return ResponseEntity.ok(feedbackService.createFeedback(account, feedbackRequest))
    }

    @ApiOperation(value = "피드백 리스트 조회 (WEB)")
    @GetMapping
    fun getFeedbackList(
        @PageableDefault pageable: Pageable
    ): ResponseEntity<FeedbackListResponse> {
        return ResponseEntity.ok(feedbackService.getFeedbackList(pageable))
    }

    @ApiOperation(value = "피드백 상세 정보 조회 (WEB)")
    @GetMapping("{feedbackId}")
    fun getFeedbackDetail(
        @PathVariable feedbackId: Long
    ): ResponseEntity<FeedbackResponse> {
        return ResponseEntity.ok(feedbackService.getFeedback(feedbackId))
    }

    @ApiOperation(value = "피드백 확인 여부 체크 하기")
    @PutMapping("/{feedbackId}")
    fun checkFeedback(
        @PathVariable feedbackId: Long,
        @RequestParam checkConfirmed: Boolean
    ): ResponseEntity<FeedbackIdResponse> {
        val account = accountService.getCurrentAccount()
        return ResponseEntity.ok(feedbackService.updateConfirmed(account, feedbackId, checkConfirmed))
    }

}